hibike/devices/LimitSwitch/limit_switch.cpp

```diff
@@ -18,43 +16,36 @@ void loop() {
   hibike_loop();
 }
 
-// you must implement this function. It is called when the device receives a DeviceUpdate packet.
-// the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_update(uint8_t param, uint32_t value) {
-  return ~((uint32_t) 0);
-}
 
-// you must implement this function. It is called when the devie receives a DeviceStatus packet.
-// the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_status(uint8_t param) {
-  return ~((uint32_t) 0);
+// You must implement this function.
+// It is called when the device receives a Device Write packet.
+// Updates param to new value passed in data.
+//    param   -   Parameter index
+//    data    -   value to write, in bytes TODO: What endian?
+//    len     -   number of bytes in data
+//
+//   return  -   size of bytes written on success; otherwise return 0
+
+uint32_t device_write(uint8_t param, uint8_t* data, size_t len){
+  return 0;
 }
 
 
-// you must implement this function. It is called with a buffer and a maximum buffer size.
-// The buffer should be filled with appropriate data for a DataUpdate packer, and the number of bytes
-// added to the buffer should be returned. 
+// You must implement this function.
+// It is called when the device receives a Device Data Update packet.
+// Modifies data_update_buf to contain the parameter value.
+//    param           -   Parameter index
+//    data_update_buf -   buffer to return data in
+//    buf_len         -   Maximum length of the buffer? TODO: Clarify
 //
-// You can use the helper function append_buf.
-// append_buf copies the specified amount data into the dst buffer and increments the offset
-uint8_t data_update(uint8_t* data_update_buf, size_t buf_len) {
+//    return          -   sizeof(param) on success; 0 otherwise
 
-  if (buf_len < sizeof(uint8_t) * NUM_SWITCHES) {
-    return 0;
-  }
+uint8_t device_read(uint8_t param, uint8_t* data, size_t len) {
 
-  uint8_t *data = (uint8_t *) data_update_buf;
-  // Read sensor
-  for (int i = 0; i < NUM_SWITCHES; i++) {
-      data[i] = 1 - digitalRead(pins[i]);  
+  if (MAX_PAYLOAD_SIZE - len < sizeof(uint8_t) || param >= NUM_SWITCHES) {
```

Comment:
Well it won't hurt.  But actually, we should remove it from everything where the param is a unit8_t (which is all of them).

Anywho, this is good for now.  THanks!